### PR TITLE
feat: entry list result data validation and tests

### PR DESCRIPTION
**Pull request by bgutsol**

### CODE REVIEW

File: lib/entry-list.ts [hunk 1]
```diff
@@ -6,6 +6,29 @@ import {
...
+export default function createEntryList(channel: Channel): EntryListAPI {
+  let _handler: OnEntryListUpdatedHandler | null = null
+
+  channel.addHandler(
+    'entryListUpdated',
+    ({ msgId, props }: { msgId: string; props: OnEntryListUpdatedHandlerProps }) => {
+      return runHandler(_handler, props).then((result: OnEntryListUpdatedHandlerReturn) => {
+        return channel.send('entryListResult', { msgId, result })
+      })
+    }
```
Comment (andipaetzold) [inline lib/entry-list.ts:18]: Nit: vs code proposes to convert this to an async function, maybe we can do that to get some newer-looking code

File: lib/entry-list.ts [hunk 2]
```diff
@@ -6,6 +6,29 @@ import {
...
+      if (!isFunction(handler)) {
+        throw new Error('OnEntryListUpdated handler must be a function')
+      } else {
+        _handler = handler
+      }
+    },
+  }
+}
+
 const isFunction = (f: any) => typeof f === 'function'
```
Comment (andipaetzold) [inline lib/entry-list.ts:32]: Can we extract this as it's also used in `app.ts`?
Or maybe inline the function directly as it's so trivial?

File: lib/entry-list.ts [hunk 3]
```diff
@@ -19,32 +42,39 @@ const runHandler = async (
...
+    return validateResult(result)
   } catch (error) {
     console.error(error)
     return false
   }
 }
 
-export default function createEntryList(channel: Channel): EntryListAPI {
-  let _handler: OnEntryListUpdatedHandler | null = null
+const validateResult = (result: OnEntryListUpdatedHandlerReturn) => {
```
Comment (andipaetzold) [inline lib/entry-list.ts:53]: It feels strange that the method returns a value which is always identical to the input or throws an error.
I'd make this an assert method that never returns anything but throws if something is wrong.

Now it feels like it's also doing some data cleanup or similar

File: lib/entry-list.ts [hunk 4]
```diff
@@ -19,32 +42,39 @@ const runHandler = async (
...
-    onEntryListUpdated(handler: OnEntryListUpdatedHandler) {
-      if (!isFunction(handler)) {
-        throw new Error('OnEntryListUpdated handler must be a function')
-      } else {
-        _handler = handler
-      }
-    },
-  }
+    if (!schema[key](data[key])) {
+      throw new Error(`EntryListResult data is invalid. Invalid value of key "${key}"`)
```
Comment (andipaetzold) [inline lib/entry-list.ts:75]:
```suggestion
      throw new Error(`EntryListResult data is invalid. Invalid value of key "${key}".`)
```

File: test/unit/entry-list.spec.ts [hunk 5]
```diff
@@ -80,6 +80,40 @@ describe('createEntryList()', () => {
 
       it('returns false when a promise rejects', () =>
         test(() => Promise.reject(new Error()), false))
+
+      it('returns false if the result data hasn`t valid key', () =>
```
Comment (andipaetzold) [inline test/unit/entry-list.spec.ts:84]:
```suggestion
      it('returns false if the result data has invalid key', () =>
```

File: test/unit/entry-list.spec.ts [hunk 6]
```diff
@@ -80,6 +80,40 @@ describe('createEntryList()', () => {
...
+            data: {
+              wrongKey: {
+                entryId: 'value',
+              },
+            } as any,
+          }),
+          false
+        ))
+
+      it('returns false if the result data hasn`t valid values type', () =>
```
Comment (andipaetzold) [inline test/unit/entry-list.spec.ts:96]:
```suggestion
      it('returns false if the result data has an invalid value type', () =>
```

**Review by andipaetzold — CHANGES_REQUESTED** [review]
Looks good. Just some nits.

**Comment by bgutsol** [comment]
> Looks good. Just some nits.

@andipaetzold thanks a lot for the feedback! 
Please check one more time to be sure I understood you right.

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**